Update Instructions for Manual Testing (#161)
AY2425S2-CS2103T-T13-1/tp: docs/DeveloperGuide.md

   1. Download the jar file and copy into an empty folder

   2. Double-click the jar file<br>
   Expected: Shows the GUI with a set of sample contacts. The window size may not be optimum.

2. Saving window preferences

   1. Resize the window to an optimum size. Move the window to a different location. Close the window.

   2. Re-launch the app by double-clicking the jar file.<br>
   Expected: The most recent window size and location is retained.

3. Exit program

>    1. Entering exit command. Close the window.

Reviewer: ```suggestion
   1. Enter the exit command. The window closes.
```

   2. Double-click the jar file<br>
   Expected: Shows the GUI with contacts. The most recent window size and location is retained.

### Deleting a client

1. Deleting a client while all clients are being shown

   1. Prerequisites: List all clients using the `list` command. Multiple clients in the list.

   2. Test case: `delete 1`<br>
      Expected: First contact is deleted from the list. Details of the deleted contact is shown in the status message.

   3. Test case: `delete x` (where x is larger than the list size)<br>
      Expected: No person is deleted. Error details of invalid client index shown in the status message.

   4. Test case: `delete 0`<br>

>       Expected: No person is deleted. Error details of incorrect format and Right usages are shown in the status message.

Reviewer: ```suggestion
      Expected: No person is deleted. Error details of incorrect format and right usage are shown in the status message.
```

   5. Other incorrect delete commands to try: `delete`, `delete y`, `...` (where y is not a positive integer)<br>
      Expected: Similar to previous.

### Displaying a client

1. Displaying a client while all clients are being shown

    1. Prerequisites: List all clients using the `list` command. Multiple clients in the list.

    2. Test case: `display 1`<br>
       Expected: First contact is displayed from the list. Details of the client is shown in the status message.

    3. Test case: `display x` (where x is larger than the list size)<br>
       Expected: No person is displayed. Error details of invalid client index shown in the status message.

    4. Test case: `display 0`<br>

>        Expected: No person is displayed. Error details of incorrect format and Right usages are shown in the status message.

Reviewer: ```suggestion
       Expected: No person is displayed. Error details of incorrect format and right usage are shown in the status message.
```

    5. Other incorrect display commands to try: `display`, `display y`, `...` (where y is not a positive integer)<br>
       Expected: Similar to previous.

### Finding a client

1. Finding a client in the client list

    1. Test case: `find x` (where x is a complete part of client's name)<br>

>     Expected: Client list only show the contacts that contains the name. Number of clients listed message is shown in the status message.

Reviewer: ```suggestion
    Expected: Client list only shows the contacts that contains the name. Number of clients listed is shown in the status message.
```

    2. Test case: `find x y` (where x and y are complete part of different client's name)<br>

>     Expected: Client list only show the contacts that contains the name. Number of clients listed message is shown in the status message.

Reviewer: ```suggestion
    Expected: Client list only shows the contacts that contains the name. Number of clients listed is shown in the status message.
```

    3. Test case: `find z` (where z is not part of any client's name)<br>

>     Expected: Client list is empty. No clients listed message is shown in the status message.

Reviewer: ```suggestion
    Expected: Client list is empty. 'No clients listed' message is shown in the status message.
```

### Saving data

1. Dealing with missing data files

   1. Prerequisites: Executed the [final FitFlow release] jar file.

   2. Open the FitFlow home folder. Delete the Data folder.

>    3. Double-click the jar file Expected: Shows the GUI with a set of sample contacts. The window size is reset.

Reviewer: ```suggestion
   3. Double-click the jar file. <br>
   Expected: Shows the GUI with a set of sample contacts. The most recent window size and location is retained.
```
Window size should remain the same since that is saved under preferences, not the data folder

    3. Test case: `display x` (where x is larger than the list size)<br>
       Expected: No person is displayed. Error details of invalid client index shown in the status message.

    4. Test case: `display 0`<br>
       Expected: No person is displayed. Error details of incorrect format and right usage are shown in the status message.

    5. Other incorrect display commands to try: `display`, `display y`, `...` (where y is not a positive integer)<br>
       Expected: Similar to previous.

### Finding a client

1. Finding a client in the client list

    1. Test case: `find x` (where x is a complete part of client's name)<br>
    Expected: Client list only shows the contacts that contains the name. Number of clients listed is shown in the status message.

    2. Test case: `find x y` (where x and y are complete part of different client's name)<br>
    Expected: Client list only shows the contacts that contains the name. Number of clients listed is shown in the status message.

    3. Test case: `find z` (where z is not part of any client's name)<br>
    Expected: Client list is empty. 'No clients listed' message is shown in the status message.

> 

Reviewer 2: Should you have the test cases where the user inputs `find` or `find ` where it is not a valid string and it displays the error details of incorrect format to keep it consistent with the other manual testing commands?
